docs/src/rules/complexity.md

```diff
@@ -148,20 +148,62 @@ function foo() { // this function has complexity = 1
 
 ## Options
 
-Optionally, you may specify a `max` object property:
+This rule has a number or object option:
 
-```json
-"complexity": ["error", 2]
-```
+* `"max"` (default `20`) enforces a maximum complexity
 
-is equivalent to
+* `"modifiedComplexity": true` use the modified complexity model
+
+### max
+
+Customize the threshold with the `max` property.
 
 ```json
 "complexity": ["error", { "max": 2 }]
 ```
 
 **Deprecated:** the object property `maximum` is deprecated. Please use the property `max` instead.
 
+Or use the shorthand syntax:
+
+```json
+"complexity": ["error", 2]
+```
+
+### modifiedComplexity
```

Comment:
I'm curious if there are other forms of cyclomatic complexity, too?

If so, it might be better to make this a string option like `form: "modified"` vs `form: "mccabe"` to allow for future expansion?

Reply:
Also good question. There seem to be at least three common implementations, in addition to the classic one. From [Understand's docs](https://support.scitools.com/support/solutions/articles/70000582297-understanding-mccabe-cyclomatic-complexity):

> **Modified Cyclomatic Complexity**: Each decision in a multi-decision structure statement (switch in C/C++) is not counted and instead the entire structure counts as 1
> **Strict Cyclomatic Complexity**: Logical ANDs and ORs in conditional expressions also add 1 to the complexity for each of their occurrences.
i.e. The statement if (a && b || c) would have a Cyclomatic of 1 but a Strict Cyclomatic of 3
> **Essential Complexity**: Iteratively replaces all well structured control structures with a single statement. Structures such as if-then-else and while loops are considered well structured.

I've also come across these in other sources, although the the ones besides the Modified Complexity are even less common. I don't have a strong opinion, but I'm slightly leaning towards keeping the `modifiedComplexity` option as is and deprecating it in favor of something like `form: "..."` if/when we want to support more models.

Reply:
I'm generally not a fan of adding things if we think there's a reasonable chance we'll need to deprecate it in the future. Deprecations cause a lot confusion and we end up needing to support more than one way of doing things for a while, which gets messy. So given that there are other forms, my vote is to go with `form`.

Reply:
Yes, I agree with the downsides of a potential deprecation. I was just trying to not over-engineer upfront. Switched to using `variant`. I like that word better than `form` in this context, and these metrics are often referred to as "variations" of (the classic) cyclomatic complexity.

Reply:
Works for me.